sycl/unittests/scheduler/QueueFlushing.cpp

```diff
@@ -225,6 +226,7 @@ TEST_F(SchedulerTest, QueueFlushing) {
       detail::QueueImplPtr TempQueueImpl = detail::getSyclObjImpl(TempQueue);
       DepEvent.reset(new detail::event_impl(TempQueueImpl));
       DepEvent->setContextImpl(TempQueueImpl->getContextImplPtr());
+      DepEvent->setStateIncomplete();
```

Comment:
I'm not positive, but I think some of these events are just the empty constructed ones and now that they are ending up here, where we actually need to wait on them, we are setting the context and marking them as incomplete.  The event_impl(q) constructor already sets these things up, but I'm fairly sure those are not the only events that the scheduler encounters.   Certainly the code could probably be restructured, but I think it's getting out of the scope of this PR.

Reply:
I agree that it may be a little out of scope. However, you do add a call to `setContextImpl` in the `event_impl` constructor that takes a queue, so for some of these (like the one here) we are doing it twice with the same context, which we should probably take care of immediately.

Reply:
See my last comment. Since some of these commands are copy commands, there is a device queue and a host queue, which may (or may not) have the same context. The correct context depends on the direction of the copy. The code just blindly uses  one of the Queues in when initializing the Command subclass, but then in the constructor does a check to make sure that blind choice was right and, if not, corrects it.

Reply:
Sure, but even in that case we first set the context to the one from the passed queue and then we set the right context afterwards. So in the end we are still following the pattern, even if it is immediately overwritten. You are right that we would need the `setContextImpl` call in the code for these Command subclasses, but the base wouldn't need it anymore.